### DISCUSSION TITLE
Dev vs Prod mode

### DISCUSSION THREAD

**Discussion by mlissner** [plausible/analytics#1766] [Anything Else]
I'm setting up Plausible for CourtListener.com and one idea I had was to set up Plausible to track page views of our developers. The idea here is that by seeing aggregate stats about how developers do their work, we can see which areas of the site suck a lot of time (are tricky to debug, absorb a lot of hours, etc) and which areas don't. 

I could see this being pretty useful to developers and product managers. I guess you could use `/etc/hosts` on all dev machines to accomplish something like this, so that you have:

```
<script
      defer
      data-domain="dev-domain.dev"
      src="https://plausible.io/js/plausible.js"></script>
```

But that's a lot to get set up on every dev machine. 

If Plausible supported this natively, you could do:

```
<script
      defer
      dev
      data-domain="courtlistener.com"
      src="https://plausible.io/js/plausible.js"></script>
```

Or via some kind of unverified domain that's given an ID by Plausible when it's first set up:

```
<script
      defer
      data-domain-id="1"
      src="https://plausible.io/js/plausible.js"></script>
```

Anyway, just a thought I had while getting things set up, that it might be nice. We use something sort of similar with Sentry so that developers can track (and see) stacktraces they get while doing dev. It's helpful!

**Comment by ukutaht** [comment]
Yeah that would be lovely. Tracking just dev vs pro mode should be pretty easy, we are planning to build some primitives that would kind of support it.

1. Custom props on pageviews: https://github.com/plausible/analytics/discussions/320

You could tag each pageview with `env: dev` or `env: prod`.

2. Saved filters: https://github.com/plausible/analytics/discussions/1495

You could save a filter named `prod` for only prod views.

3. Default filter

Once we have saved filters (I guess you could call them 'views' like GA), then you can set 'prod' as your default. Boom! You have dev and prod separation with the default dashboard showing the real data you care about most of the time. And by changing the view or default filter, you can see the dev stuff.


What is complicated though is if you want each dev's data to be separate. That's more difficult but could also be done with custom props

**Reply by mlissner** [reply]
Interesting approach. I'll watch for those new features. 

> What is complicated though is if you want each dev's data to be separate.

FWIW, I think I would *not* want that for our devs. I think that'd put us back in tracking land in an uncomfortable way. :)

**Reply by ukutaht** [reply]
> FWIW, I think I would not want that for our devs. I think that'd put us back in tracking land in an uncomfortable way. :)

Totally :)

**Reply by robinpecha** [reply]
> 2. Saved filters: [Extend shared links with filters #1495](https://github.com/plausible/analytics/discussions/1495)
> You could save a filter named `prod` for only prod views.
Hi, thanks for the instructions. But filters don't work on a domain, just on a path. Not even with wildcards. 
I tried: https://mypage.dev or https://mypage.com. Nor \*\*mypage.dev\*\* works.
I'm doing something wrong?

**Comment by ukutaht** [comment]
@mlissner  seeing as CourtListener.com has a search box in the hero section, I'd love to get your feedback on this as well: https://github.com/plausible/analytics/discussions/164

A very basic version of site search is planned soon so it's a good time to have that discussion

**Reply by mlissner** [reply]
I took a look. We have Matomo now, but I don't use the site search feature beyond just glancing at it every few years. As a legal website, we have tens of millions of searchable items, so on most days, there aren't many discernible trends in our search queries. 

I'm afraid I don't have much to add to the conversation, but I'll definitely try the feature and see if it's useful to us once it's ready.

**Comment by robinpecha** [comment]
We have different domains for prod/dev so this code solves our problem:
https://github.com/plausible/analytics/discussions/482

```
<script>
var script = document.createElement('script')
script.async=true
script.defer=true
script.dataset.domain=window.location.host // Set script data-domain dynamically
script.src = "https://plausible.io/js/plausible.js"
document.head.append(script)
</script>

```
